src/EventListener/InsertTagsListener.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Terminal42\ShortlinkBundle\EventListener;
+
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Terminal42\ShortlinkBundle\Entity\Shortlink;
+use Terminal42\ShortlinkBundle\Repository\ShortlinkRepository;
+use Terminal42\ShortlinkBundle\ShortlinkGenerator;
+
+/**
+ * @Hook("replaceInsertTags")
+ */
+class InsertTagsListener
+{
+    private ShortlinkGenerator $generator;
+    private ShortlinkRepository $repository;
+
+    public function __construct(ShortlinkGenerator $generator, ShortlinkRepository $repository)
+    {
+        $this->generator = $generator;
+        $this->repository = $repository;
+    }
+
+    public function __invoke(string $tag)
+    {
+        $chunks = explode('::', $tag);
+
+        if ('shortlink' === $chunks[0]) {
+            /** @var Shortlink $link */
+            $link = $this->repository->find((int) $chunks[1]);
+
+            if (null === $link) {
+                return '';
+            }
+
+            return '//'.$this->generator->generateFromEntity($link);
```

Comment:
We should really get rid of relative URLs. Even more so of protocol relative URLs. Why not just use an absolute link, it makes everything so much easier?

Reply:
How we should determine whether it's `http://` or `https://` in this case? We can't base it on the current request because the host is configurable in the `config.yml`, so we would have to set it there, no? 🤔

Reply:
using a protocol relative link here basically means the same as using the protocol of the current request. So maybe the generator should always return a real url?